### PR TITLE
Integrate VAE Decoder into SDv1-4 demo

### PR DESCRIPTION
**Pull request by astancovTT**
### Ticket
[#21688](https://github.com/tenstorrent/tt-metal/issues/21688)
### Problem description
Currently host VAE is used for SDv1-4

### What's changed
- Integrated on device VAE Decoder into Stable Diffusion v1-4 demo

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/15016317262)
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes
- [x] [(Single-card) Frequent model and ttnn tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml) CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/15019651491)

### CODE REVIEW

**Review by s-jovic — CHANGES_REQUESTED** [review]

File: models/demos/wormhole/stable_diffusion/demo/demo.py [hunk 1]
```diff
@@ -551,11 +577,22 @@ def run_demo_inference_diffusiondb(
         latents = ttnn.to_torch(ttnn_latents).to(torch.float32)
         # scale and decode the image latents with vae
         latents = 1 / 0.18215 * latents
-        image = vae.decode(latents).sample
+
+        # on blackhole, we use the original vae decoder until #20760 is fixed
+        if not is_blackhole():
+            latents_tt = ttnn.from_torch(
+                latents.permute([0, 2, 3, 1]), device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16
```
[the remark this answers is not in the document]
Comment (astancovTT) [reply models/demos/wormhole/stable_diffusion/demo/demo.py:584]: Will keep latents on device, but the `NCHW` shape is needed for `scheduler.step`

**Review by s-jovic — APPROVED** [review]
